### ISSUE TITLE
Encoding ground-truth labels for training is extremely slow.

### ISSUE THREAD

**Issue by jrobchin** [pierluigiferrari/ssd_keras#131]
I'm trying to train the SSD7 on the [2DMOT2015 Dataset](https://motchallenge.net/data/2D_MOT_2015/) and the encoding step of the [ssd_input_encoder.py](https://github.com/pierluigiferrari/ssd_keras/blob/master/ssd_encoder_decoder/ssd_input_encoder.py) takes a few minutes per batch of 16 images (300, 300, 3). The actual training step takes a couple seconds. Some images have more than 25 bounding boxes in a single frame and I'm wondering if that could be an issue.

I'm currently using the train_generator and the fit_generator() method so encoding happens between training steps. I've already invesitgated the encoder itself and the encoding step is definitely the bottleneck. I've also loaded all images into memory before training. I'm sure it's the generator because I've used next() on it and it takes a long time for even a batch size of 16 images.

Would it be possible to encode the data before hand and export it to a file so it does not have to be computed at every step?

Hardware should not be an issue as training is running on a GTX 1080 and utilization only goes above 0% for a couple seconds when the actual computations happen.

Thanks in advance!

**Comment by pierluigiferrari** [comment]
"A few minutes per batch" does sound crazy and I haven't done the math yet, but the runtime complexity of the bipartite matching process is quadratic in the number of ground truth boxes (with a pretty large multiplicative constant, the number of anchor boxes).

I'll download that dataset and try it myself when I find the time to see whether I can confirm your numbers.
